packages/test-utils/src/testUtils/apis/StorageApi/MockStorageApi.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Observable,
+  StorageApi,
+  storageApiRef,
+  StorageValueChange,
+} from '@backstage/core-api';
+import ObservableImpl from 'zen-observable';
+
+export class MockStorageApi implements StorageApi {
+  static factory = {
+    implements: storageApiRef,
+    deps: {},
+    factory: () => new MockStorageApi(''),
+  };
+
+  private readonly namespace: string;
+  private readonly data: Map<string, any>;
+
+  constructor(namespace: string, data?: Map<string, any>) {
```

Comment:
Wonder if we could switch this around or smth? Doesn't really make sense to give a namespace when you're supplying the mock data. Would also be convenient if you can give an object instead. Maybe make it private and add `MockStorageApi.create`?

Reply:
Ah but it's not meant for mock data specifically - it's just used for making sub-buckets, you would normally never give this parameter yourself

Reply:
yeah that's why I think it's best to not make it part of the API

Reply:
Hm yes but this is not a public api as such at all. It's just for use by the automatic wrapper basically. And we need to be able to pass an instance of the store when creating a bucket. How do you propose that it be structured? I would have liked to make the constructor private actually

Reply:
It should be public API. The APIs in the mock registry is for when you don't care about the mock, when you essentially just want an empty implementation. If you want to test your components interaction with the storage API, the idea is that you'd do this:

```tsx
const storageApi = MockStorageApi.create({ 'ui/MyComponent/my-key/hide-banner': true });

const rendered = renderInTestApp(
  <ApiProvider apiRef={storageApiRef} api={storageApi}>
    <MyComponent storageKey="my-key">
  </ApiProvider>
)

expect(rendered.queryByTestId('banner')).not.toBeInTheDocument();
```

Reply:
Yarrp, made it a create method and private constructor instead